src/write.rs

```diff
@@ -133,6 +134,13 @@ impl<W: Write + io::Seek> Write for ZipWriter<W> {
                 "No file has been started",
             ));
         }
+
+        // Check if the buffer is text (UTF8) or contains binary data. Note that if an UTF8
+        // character is split between to calls to write, this will falsly mark it as binary.
+        // For ASCII characters, this is always correct.
```

Comment:
Can we use `Utf8Error::valid_up_to` to figure out if we’re in this situation? I think to know for sure we would have to keep the bytes from any previous read the are potentially a non-complete character and if validation fails, prepend and check again.

Reply:
We could also just mark it as text for ASCII files. The appnote is a bit vague here:
```
       4.4.14.1 The lowest bit of this field indicates, if set, 
       that the file is apparently an ASCII or text file.  If not
       set, that the file apparently contains binary data.
```